upload.go

```diff
@@ -0,0 +1,93 @@
+package malwarebazaar
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"time"
+)
+
+// UploadFile takes the file contents passed any uploads it to malware bazaar with the given fields set. This function doesn't have the ability to set comments or context for the sample
+func UploadFile(ctx context.Context, filename string, contents []byte, anonymous bool, tags []string, deliveryMethod string, apikey string) ([]byte, error) {
+	httpCli := http.Client{
+		Timeout: 15 * time.Second,
+	}
+
+	if filename == "" {
+		filename = fmt.Sprintf("%x", md5.Sum(contents))
+	}
+
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	fw, err := writer.CreateFormFile("file", filename)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = io.Copy(fw, bytes.NewBuffer(contents))
```

Comment:
If you're copying anyway, you should accept an `io.Reader` instead of accepting a `[]byte`.  That way a caller can pass a direct reader instead of necessarily storing the entire file in memory.
You know how I like my readers ;)

Reply:
fixed